storage/innobase/xtrabackup/src/xbcrypt.c

```diff
@@ -533,26 +554,19 @@ mode_encrypt(File filein, File fileout)
 				continue;
 			}
 
-			if (encryptbuflen < bytesread) {
-				if (encryptbuflen) {
-					encryptbuf = my_realloc(encryptbuf,
-								bytesread,
-								MYF(MY_WME));
-					encryptbuflen = bytesread;
-				} else {
-					encryptbuf = my_malloc(bytesread,
-							       MYF(MY_WME));
-					encryptbuflen = bytesread;
-				}
+			if (encryptbuflen < origbuflen) {
+				encryptbuf = my_realloc(encryptbuf, origbuflen,
+					MYF(MY_WME | MY_ALLOW_ZERO_PTR));
```

Comment:
`MY_WME` another flag which tells to print error message and continue in case of memory allocation error. Actually, I cloud replace it with `MY_FAE`.
`MY_ALLOW_ZERO_PTR` is a flag telling `my_realloc` to handle NULL pointer correctly if it has been passed. I.e. I allows to avoid construction `if (ptr == NULL) ptr=malloc() else ptr=realloc()`